### PR TITLE
feat(UIForms): datalist text mode

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
PR #1625 introduced the forms text display mode.
This PR implements text mode for Datalist.
- text display
- async title map loading

**What is the chosen solution to this problem?**
- extract a common part for trigger management
- call `didmount` and `focus` triggers to load async title map
- register `datalist_text` in widget dictionary

This is testable via surge > forms > core-datalist

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -17,6 +17,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/CheckBox/SimpleCheckBox.component.js
   9:4  error  Form controls using a label to identify them must be programmatically associated with the control using htmlFor  jsx-a11y/label-has-for
 
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/fields/Datalist/Datalist.component.js
+  65:2  error  getTitleMap should be placed after onTrigger  react/sort-comp
```
Comment (talend-glorieux) [inline output/forms.eslint.txt]: Remove that error.

File: packages/forms/src/UIForm/fields/Datalist/Datalist.trigger.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+			const newState = data.reduce((state, nextData) => ({ ...state, ...nextData }), {});
+			onResponse(newState);
+			onLoading(false);
+		},
+		error => {
+			onError(error);
+			onLoading(false);
+		},
+	);
+}
```
Comment (talend-glorieux) [inline packages/forms/src/UIForm/fields/Datalist/Datalist.trigger.js]: I don't understand that function. :thinking:

File: packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		const onTrigger = (event, trigger) =>
+			this.props.onTrigger(event, {
+				trigger,
+				schema: this.props.schema,
+				errors: this.props.errors,
+				properties: this.props.properties,
+			});
+
+		callTrigger(null, {
+			eventNames: ['didMount', 'focus'],
```
Comment (talend-glorieux) [inline packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js]: Should be shared constants.

File: packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		return (
+			<FieldTemplate id={id} label={title}>
+				{value && this.state.isLoading ? (
+					<span className={this.state.isLoading ? theme.loading : undefined} aria-busy="true">
+						{value} {`(${t('DATALIST_WIDGET_LOADING_LABELS', { defaultValue: 'loading labels' })})`}
+					</span>
+				) : (
+					(titleEntry && titleEntry.name) || value
+				)}
+			</FieldTemplate>
```
Comment (talend-glorieux) [inline packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js]: Should be a simple `if/else`. It also looks like `this.state.isLoading` is tested twice.

File: packages/forms/src/UIForm/fields/Datalist/Datalist.trigger.js (hunk 2 above)
Comment (jsomsanith) [reply packages/forms/src/UIForm/fields/Datalist/Datalist.trigger.js]: Added a jsDoc.

File: packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js (hunk 4 above)
Comment (jsomsanith) [reply packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js]: `It also looks like this.state.isLoading is tested twice.` --> thank you that survived my refacto :/

**Pull request merged**